src/safeposix/syscalls/net_calls.rs

```diff
@@ -3075,43 +3189,105 @@ impl Cage {
             );
         }
         return 0;
     }
 
+    /// ## ------------------EPOLL_WAIT SYSCALL------------------
+    /// ### Description
+    /// The epoll_wait_syscall waits for events on the epoll instance
+    /// referred to by the file descriptor epfd. The buffer pointed to by events
+    /// is used to return information from the ready list about file descriptors
+    /// in the interest list that have some events available.  Up to maxevents
+    /// are returned by epoll_wait_syscall(). The maxevents argument must be
+    /// greater than zero.
+
+    /// ### Function Arguments
+    /// The `epoll_wait_syscall()` receives four arguments:
+    /// * `epfd` - the epoll file descriptor to be applied the action
+    /// * `events` - The buffer of array of EpollEvent used to store returned
+    ///   information from the ready list about file descriptors in the interest
+    ///   list that have some events available
+    /// * `maxevents` - maximum number of returned events. The maxevents
+    ///   argument must be greater than zero.
+    /// * `timeout` - The timeout argument is a RustDuration structure that
+    ///   specifies the interval that epoll_wait_syscall should block waiting
+    ///   for a file descriptor to become ready.
+
+    /// ### Returns
+    /// On success, epoll_wait_syscall returns the number of file descriptors
+    /// ready for the requested I/O operation, or zero if no file descriptor
+    /// became ready when timeout expires
+    ///
+    /// ### Errors
+    /// * EBADF - epfd is not a valid file descriptor.
+    /// * EINTR - The call was interrupted by a signal handler before either (1)
+    ///   any of the requested events occurred or (2) the timeout expired
+    /// * EINVAL - epfd is not an epoll file descriptor, or maxevents is less
+    ///   than or equal to zero.
+    ///
+    /// ### Panics
+    /// * when maxevents is larger than the size of events, index_out_of_bounds
+    ///   error would be raised
     pub fn epoll_wait_syscall(
         &self,
         epfd: i32,
         events: &mut [EpollEvent],
         maxevents: i32,
         timeout: Option<interface::RustDuration>,
     ) -> i32 {
+        // current implementation of epoll is still based on poll_syscall,
+        // we are essentially transform the epoll input to poll input then
+        // feed into poll_syscall, and transform the poll_syscall output
+        // back to epoll result. Such method gives several issues:
+        // 1. epoll is supposed to support a brand new mode called edge-triggered
+        // mode, which only consider a fd to be ready only when new changes are made
+        // to the fd. Currently, we do not support this feature
+        // 2. several flags, such as EPOLLRDHUP, EPOLLERR, etc. are not supported
+        // since poll_syscall currently do not support for these flags, so epoll_syscall
+        // that rely on poll_syscall would subsequently not able to support them
```

Comment:
so epoll_syscall that relies on poll_syscall, as a consequence, does not support them*